### PR TITLE
fix: turn off incremental builds

### PR DESCRIPTION
**Pull request by aatifsyed**
## Summary of changes
- Multiple team members are hitting ICEs due to incremental builds
- This disables that for our repo
- Note you can still enable this on your machine if it's important for you, by adding config in a parent directory: https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure

### CODE REVIEW

File: .github/workflows/forest.yml
Comment (ruseinov) [inline .github/workflows/forest.yml:1]: curious: Was it also a problem in CI?

Comment (lemmih) [reply .github/workflows/forest.yml:1]: Nope. The CI already disables incremental builds.

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — COMMENTED** [review]
Is it actually disabled? To my understanding, if you omit this entry, the build will default to the profile value, which is `false` only for release builds.

What is the compilation time impact of this change?

**Review by elmattic — APPROVED** [review]

**Comment by elmattic** [comment]
> Is it actually disabled? To my understanding, if you omit this entry, the build will default to the profile value, which is `false` only for release builds. That is to say, incremental compilation will still be there for the dev builds.

We can disable it with:
```
[profile.dev]
incremental = false
```

That said I would prefer to leave this choice to the developer. Safe goes for the compilation backend in debug (i.e. cranelift etc).

**Comment by ruseinov** [comment]
> That said I would prefer to leave this choice to the developer. Same goes for the compilation backend in debug (i.e. cranelift etc).

I tend to agree. Why don't we leave the flag be and just make it `false` by default? It's way easier to work with than having to re-introduce the whole section.

**Comment by aatifsyed** [comment]
> One other solution to all these problems is to remove config.toml from the repository altogether and git ignore .cargo dir. That would be more idiomatic.

I like this

**Comment by aatifsyed** [comment]
https://github.com/ChainSafe/forest/issues/3146

**Pull request closed without merging**